core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -387,13 +384,6 @@ object SparkSubmit {
     }
 
     if (isYarnCluster) {
-      // In yarn-cluster mode for a python app, add primary resource and pyFiles to files
-      // that can be distributed with the job
-      if (args.isPython) {
-        args.files = mergeFileLists(args.files, args.primaryResource)
-        args.files = mergeFileLists(args.files, args.pyFiles)
-      }
-
       // In yarn-cluster mode for a R app, add primary resource to files
       // that can be distributed with the job
       if (args.isR) {
```

Comment:
now that you removed the python stuff you can simplify this to `if (isYarnCluster && args.isR) {`